dotcom-rendering/docs/architecture/028-dotenv.md

```diff
@@ -38,3 +38,27 @@ const secrets = [
     // ...
 ];
 ```
+
+## Security
+
+`.env` variables are only available on the server, you should ensure that you **never** pass an environment variable into the clientside code
+e.g
+
+```jsx
+<Island>
+	<!-- Bad :((( -->
+	<MyComponent myProp={process.env.MY_VAR} />
+</Island>
+```
```

Comment:
Maybe we could give a good example here e.g. using the env var to pass a salted URL to the component?

What happens if we read a process.env in an `.importable`?

```suggestion
```jsx
return (
    <Island>
	    <!-- Bad :((( -->
	    <Image imageSalt={process.env.IMAGE_SALT} />
    </Island>
)

const url = getSaltedUrl(process.env.IMAGE_SALT)
return (
    <Island>
	    <!-- Good :))) -->
	    <Image url={url} />
    </Island>
)
```
```

Reply:
> What happens if we read a process.env in an .importable

On the client this will simply be undefined (and perhaps cause an error) and on the server is should work as expected

Reply:
Worth adding the good example so people know how they might do it?